### PR TITLE
Use cp -Rp for the literal install

### PR DESCRIPTION
**Pull request by TAAPArthur**
Instead of calling copy N times for each file, just copy the entire
directory with -R flag. Copy each top level dir explicitly so /etc can be
excluded.

Also instead of parsing the output of ls to get the desired permissions,
just use the -p flag to preserve them.

### CODE REVIEW

**Comment by dilyn-corner** [comment]
>Also instead of parsing the output of ls to get the desired permissions,
just use the -p flag to preserve them.

I believe the problem you will face here is that packages won't be installed with proper permissions -- all of the files when a package is built and packaged are owned by the user who ran `kiss`.

**Comment by TAAPArthur** [comment]
@dilyn-corner Good catch. Is it fair to assume the owner should be root? If so, we could just `chown -R root:root` the temp dir we are copying from.

**Comment by dilyn-corner** [comment]
I honestly don't remember why @dylanaraps didn't take this "obvious" route?

**Comment by dilyn-corner** [comment]
Thinking on this a little, the obvious reason is if a build script chowns files itself. For instance, `qemu` might want to make things owned by the kvm group.

**Comment by TAAPArthur** [comment]
But at the time of install `qemu` can't assume the `kvm` group exists. The only users/groups it can assume exists are root:root. 

Did quick check over community and don't see any explicit chown in the build script. I see some in the some post-install script which are irrelevant to the issue at hand.

**Comment by dilyn-corner** [comment]
At any rate, the kvm group can probably be safely assumed on KISS, though I'm uncertain what standard defines a default `/etc/group`. 

But you're right; I missed this singular line when reading through the code: 
https://github.com/kiss-community/kiss/blob/adf781840e85bbd61811c36567e3dc4b130c01be/kiss#L1136

So clearly... no.

I don't think `chown -R` is POSIX?

**Comment by TAAPArthur** [comment]
`chown -R` is in sbase and generally take sbase to be the lowest common denominator. It is also mentioned https://pubs.opengroup.org/onlinepubs/9699919799/utilities/chown.html

In regards to certain groups being assumed... I for one try to remove unused groups so have fewer than what the default baselayout provides. And while forking is always an option, do think we should avoid assuming certain groups exist. Or at least have scripts not fail. Just my two cents.

**Comment by dylanaraps** [comment]
To rid ourselves of fakeroot (which is awful, no understatement), the package manager creates tarballs with `$USER` permissions and then "fixes" these by using `chown root:root` during the installation process. This works fine in practice for 99% of packages. The workaround for packages needing more control has always been to leverage the `post-install` hook. Debian are swapping to "our" method (though they likely started in this direction before we existed ;)).

This same rigidness can be seen with the package manager's handling of setuid/setgid which are stripped from permissions on file installation. This is to prevent issues where a setgid directory "infects" the entire rootfs. ie, user `$HOME` contains setgid, this applies to all nested paths, user builds package with kiss, setgid infects tarball, user installs package, setgid infects `/` and thus the whole system. 

I'm experimenting locally with using `pax` to do the copy (whole directory at once). This should implement the behavior you desire. Otherwise, it might be more robust in the long run to depend once again on `rsync` (or some equivalent) for the heavy lifting. Installation speeds have always scaled badly.

Basically: There's a lot of nuance to package installation and it's hard to handle all edge cases without being *slow*.

**Comment by dilyn-corner** [comment]
@TAAPArthur - I felt that `chown -R` not being POSIX was wrong, but (I have no idea where I was looking) I didn't see it.

It looks like switching to `pax` solves quite a few problems. I'll start testing it :v

**Comment by dilyn-corner** [comment]
Looks like this has been abandoned...?
